core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/GetWalksHandler.java

```diff
@@ -0,0 +1,147 @@
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * An operation handler for {@link GetWalks} operations.
+ * <p>
+ * Currently the handler only supports creating {@link Walk}s which contain
+ * {@link Edge}s.
+ */
+public class GetWalksHandler implements OutputOperationHandler<GetWalks, Iterable<Walk>> {
+
+    private int hops;
+
+    @Override
+    public Iterable<Walk> doOperation(final GetWalks getWalks, final Context context, final Store store) throws OperationException {
+
+        // Check input
+        if (null == getWalks.getInput()) {
+            return null;
+        }
+
+        // Check operations input
+        final Iterable<GetElements> operations = getWalks.getOperations();
+
+        if (null == operations || Iterables.isEmpty(operations)) {
+            return new EmptyClosableIterable<>();
+        }
+
+        hops = getWalks.getOperations().size();
+
+        final List<AdjacencyList<Object, Edge>> adjacencyLists = new ArrayList<>();
+
+        List<Edge> results = null;
+
+        // Execute the GetElements operations
+        for (final GetElements getElements : operations) {
+            if (null == results) {
+                getElements.setInput(getWalks.getInput());
+
+                // Cache the results locally
+                results = getLimitedResults(store, context, getWalks, getElements);
+            } else {
+                final OperationChain<CloseableIterable<? extends Element>> opChain = new OperationChain.Builder()
+                        .first(new ToVertices.Builder()
+                                .input(results)
+                                .useMatchedVertex(ToVertices.UseMatchedVertex.OPPOSITE)
+                                .build())
+                        .then(new ToEntitySeeds())
+                        .then(getElements)
+                        .build();
+
+                // Cache the results locally
+                results = getLimitedResults(store, context, getWalks, opChain);
+            }
+
+            final AdjacencyList<Object, Edge> adjacencyList = new AdjacencyList<>();
+
+            // Store results in an AdjacencyList
+            for (final Edge e : results) {
+                adjacencyList.put(e.getMatchedVertexValue(), e.getAdjacentMatchedVertexValue(), e);
+            }
+
+            adjacencyLists.add(adjacencyList);
+        }
+
+        // Track/recombine the edge objects and convert to return type
+        return Streams.toStream(getWalks.getInput())
+                .map(seed -> walk(seed.getVertex(), null, adjacencyLists, new LinkedList<>()))
+                .flatMap(List::stream)
+                .collect(Collectors.toList());
+    }
+
+    private List<Walk> walk(final Object curr, final Object prev, final List<AdjacencyList<Object, Edge>> adjacencyLists, final LinkedList<Set<Edge>> queue) {
+        final List<Walk> walks = new ArrayList<>();
+
+        if (null != prev && hops != queue.size()) {
```

Comment:
Perhaps store queue.size() as a variable called hop. To help document that you are using it to calculate which hop you are looking at.

Reply:
This is difficult to do since the size of the queue can change after that variable would have been assigned (we'd have to track all changes to the size of the queue). I'll add some documentation to make this clearer.